cirq/google/engine/engine_job.py

```diff
@@ -365,6 +368,9 @@ def _raise_on_failure(job: quantum.types.QuantumJob) -> None:
     def __iter__(self) -> Iterator[study.TrialResult]:
         return iter(self.results())
 
+    def __getitem__(self, item) -> study.TrialResult:
+        return list(self.results())[item]
```

Comment:
I we believe type annotations then `results` returns a list, so we shouldn't need to invoke the constructor.

```python
def results(self) -> List[study.TrialResult]:
```

Reply:
oop fixed.

Reply:
fixed